src/widgets/details/factory.rs

```diff
@@ -337,8 +353,19 @@ impl AsyncFactoryComponent for TaskDetailsFactoryModel {
 				.task
 				.reminder_date
 				.map(|date| DateTime::from(date).format("%m/%d/%Y").to_string()),
+			sub_tasks: FactoryVecDeque::new(
+				adw::PreferencesGroup::default(),
+				sender.input_sender(),
+			),
 			dirty: false,
+		};
+		for sub_task in init.task.sub_tasks {
+			model
+				.sub_tasks
+				.guard()
+				.push_back(SubTaskInit::new(sub_task));
```

Comment:
It's more efficient to get the guard once instead of creating it and dropping it every iteration.

Reply:
Right as always! 😉